src/aleph/tcp.clj

```diff
@@ -204,13 +209,16 @@
                                        (netty/ssl-handler (.channel pipeline) ssl-context remote-address ssl-endpoint-id-alg)))
                            (.addLast pipeline "handler" handler)
                            (when pipeline-transform
-                             (pipeline-transform pipeline)))]
-    (-> (netty/create-client-chan
-          {:pipeline-builder    pipeline-builder
-           :bootstrap-transform bootstrap-transform
-           :remote-address      remote-address
-           :local-address       local-address
-           :transport           (netty/determine-transport transport epoll?)
-           :connect-timeout     connect-timeout})
-        (d/catch' #(d/error! s %)))
+                             (pipeline-transform pipeline)))
+        ch-d (netty/create-client-chan
+              {:pipeline-builder    pipeline-builder
+               :bootstrap-transform bootstrap-transform
+               :remote-address      remote-address
+               :local-address       local-address
+               :transport           (netty/determine-transport transport epoll?)
+               :connect-timeout     connect-timeout})]
+    (d/catch' ch-d #(d/error! s %))
+    (d/catch' s (fn [e]
+                  (d/error! ch-d e)
+                  (d/error-deferred e)))
```

Comment:
Again, that final `(d/error-deferred e)` is disappearing into the ether. Nothing binds the result of `(d/catch' s (fn [e] ...`

Reply:
See https://github.com/clj-commons/aleph/pull/721#discussion_r1561040043 (resolving this one already).